xbmc/epg/EpgDatabase.cpp

```diff
@@ -168,11 +168,29 @@ bool CEpgDatabase::Delete(const CEpgInfoTag &tag)
   return DeleteValues("epgtags", filter);
 }
 
-int CEpgDatabase::Get(CEpgContainer &container)
+int CEpgDatabase::GetAll(CEpgContainer &container)
 {
   int iReturn(-1);
+  int nEpgMax(0); 
 
-  std::string strQuery = PrepareSQL("SELECT idEpg, sName, sScraperName FROM epg;");
+  CEpg epg;
+  
+  /* Get number of Epgs for Progressbar */
+  std::string strQuery = PrepareSQL("SELECT count(*) FROM epg;");
+  if (ResultQuery(strQuery))
+  {
+    nEpgMax = m_pDS->fv(0).get_asInt();
+    m_pDS->close();
+  }
+
+  strQuery = PrepareSQL("SELECT epg.idEpg, epg.sName, epg.sScraperName, \
+    epgtags.idEpg, epgtags.iStartTime, epgtags.iEndTime, epgtags.iFirstAired, \
+	epgtags.iBroadcastUid, epgtags.idBroadcast, epgtags.sTitle, epgtags.sPlotOutline, \
+    epgtags.sPlot, epgtags.sOriginalTitle, epgtags.sCast, epgtags.sDirector, epgtags.sWriter, \
+    epgtags.iYear, epgtags.sIMDBNumber, epgtags.iGenreType, epgtags.iGenreSubType, epgtags.sGenre, \
+    epgtags.iParentalRating, epgtags.iStarRating, epgtags.bNotify, epgtags.iEpisodeId, epgtags.iEpisodePart, \
+    epgtags.sEpisodeName, epgtags.iSeriesId, epgtags.sIconPath \
+    FROM epg left join epgtags on (epg.idEpg = epgtags.idEpg) order by epg.idEpg;");
```

Comment:
i would prefer that you use quotes at the end instead of \.
And if you want to fetch all columns of epgtags use `epgtags.*` in the select list.

```
"SELECT epg.idEpg, epg.sName, epg.sScraperName, "
"epgtags.idEpg, epgtags.iStartTime, epgtags.iEndTime, epgtags.iFirstAired, "
...
```